CMakeLists.txt

```diff
@@ -737,3 +753,45 @@ string(APPEND _str
 # Output to stdout
 execute_process(COMMAND ${CMAKE_COMMAND} -E echo "${_str}")
 set(_str)
+
+#
+# Write a basic modulefile
+#
+set(LBANN_MODULEFILE_NAME "lbann-${LBANN_VERSION}.lua"
+  CACHE STRING
+  "The name of the LBANN modulefile to install. Must end in .lua.")
+
+if (NOT (LBANN_MODULEFILE_NAME MATCHES ".+\.lua"))
+  message(WARNING
+    "LBANN_MODULEFILE_NAME must have extension \".lua\". Appending.")
+  set(LBANN_MODULEFILE_NAME "${LBANN_MODULEFILE_NAME}.lua"
+    CACHE STRING "" FORCE)
+endif ()
+
+# TODO (trb) -- describe the configuration at all?
+# TODO (trb) -- move elsewhere? This is getting messy.
+
+set(_LBANN_MODULE_TEXT
+  "
+help(
+[[
+LBANN version ${LBANN_VERSION}. Livermore Big Artificial Neural Network
+Toolkit. A distributed memory, HPC-optimized, model and data parallel
+training toolkit for deep neural networks.
+]])
+
+whatis(\"Package: LBANN\")
+whatis(\"Version: ${LBANN_VERSION}\")
+whatis(\"Description: Livermore Big Artificial Neural Network Toolkit. A distributed memory, HPC-optimized, model and data parallel training toolkit for deep neural networks.\")
+whatis(\"URL: https://github.com/llnl/lbann\")
+
+prepend_path(\"PATH\",\"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}\")
+prepend_path(\"LD_LIBRARY_PATH\",\"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}\")
+prepend_path(\"PYTHONPATH\",\"${PYTHON_INSTALL_PREFIX}/${CMAKE_INSTALL_PYTHONDIR}\")
+
+pushenv(\"LBANN_DIR\",\"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DIR}\")
+")
+file(WRITE "${CMAKE_BINARY_DIR}/${LBANN_MODULEFILE_NAME}.install" "${_LBANN_MODULE_TEXT}")
```

Comment:
Would this be better with `configure_file`?

Reply:
Yeah probably. It just seemed so short I didn't know if I wanted to bother. But you're probably right.

Reply:
I have changed it to use a configure file. I took the opportunity to dump some config information into the comments of the module file. This isn't necessary, but might be useful for folks trying to remember what they did (spack dumps the entire CMake line into their modules).

You can get the path to the lua file from `ml show <modulename>` and then read that file (`ml show` strips comments). If it would be useful, we could also dump these configuration things as part of the help string or the whatis information, but this seemed like overkill for this.

Reply:
It seemed like overkill, but @bvanessen liked it in the whatis information so it's in the whatis information. `ml whatis lbann-0.99.0`